src/task/future.rs

```diff
@@ -85,6 +88,23 @@ unsafe fn waker(task: *const Task) -> Waker {
 #[inline]
 unsafe fn clone_raw(this: *const ()) -> RawWaker {
     let task_cell = clone_task(this as *const Task);
+    let extras = { &mut *task_cell.0.extras.get() };
```

Comment:
I think we should use a immutable reference here and get the mutable reference only at L98 to make the program sound.

Reply:
Why is it unsound?

Reply:
Two wakers in different threads concurrently, then there can be two multable references to the extras at the same time.

Reply:
If you mean `clone_raw` can be executed concurrently due to L100, then it's unsound either to move it to L98. Lock is required.

Reply:
Ahh, you're right. Executing L98 and L91 concurrently is unsound.